### ISSUE TITLE
Bug: mux-player-react sends hundreds of 0 chunk calls per second until crash or user presses play

### ISSUE THREAD

**Issue by ghost** [muxinc/elements#688]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Which Mux Elements/Packages does this apply to? Select all that apply

mux-player-react

### Which browsers are you using?

Chrome

### Which operating systems are you using?

macOS

### Description

I am using `"@mux/mux-player-react": "^1.11.0"` with `"next": "v13.4.0"`. When I run locally, my mux video and next app look fine. When I deploy to vercel and look at the console, I see about 100 requests a second against a mux chunk service. It looks like it is seeking the 0th chunk. It will continue to make tens of thousands of calls

[Image showing the Chrome network tab with 800+ requests](https://i.ibb.co/Hzjq5Bm/Screenshot-2023-05-09-at-1-56-11-PM.png)

The second I press the play button, the video will start loading and the chunk requests proceed normally from 0 to however many chunks there are. In other words, pressing play eliminates the issue. 

This problem only occurs with signed URLs. 

### Reduced test case

https://github.com/sharpsteelsoftware1/aob-web/blob/main/app/media/module/components/MediaPlayer.tsx
https://github.com/sharpsteelsoftware1/aob-web/blob/main/app/media/%5Bslug%5D/page.tsx
https://adventuresofbubbles.com/media/st-louis-central-station

### Steps to reproduce

1. Codebase is listed above, please see link
2. Go to https://adventuresofbubbles.com/media/st-louis-central-station
3. See there are 100+ network calls a second to a Mux endpoint in Chrome network inspector
4. Press play and notice that the infinite 0 chunk calls have stopped. 

### Expected Behavior

The 0th chunk should not be repeat called thousands of times.

**Comment by sharpsteelsoftware1** [comment]
I noticed that the thumbnail image is not loading in production vercel. I am using signed URLs. I think the signed url thumbnail fetch is failing infinitely.

**Comment by gkatsev** [comment]
I can reproduce this using our example page as well (https://elements-demo-nextjs-mux.vercel.app/MuxPlayer). Weirdly, as you mentioned, running the local version doesn't have the same issue.

Also, it seems like it doesn't occur in the mux-player web component directly, only via the react version.

**Comment by gkatsev** [comment]
One workaround would be to set preload=none, which means no data will be loaded until a user requests playback.

[Example using your content](https://elements-demo-nextjs-mux.vercel.app/MuxPlayer?preload=%22none%22&tokens=%7B%22playback%22%3A%22eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImNCMjRZNUdiQm1DeXZTS2hEOW5FZXhaZDMzZ0lTZk80VVUzV2Q5cXM0aXcifQ.eyJleHAiOjE2ODM3NTUwMDEsImF1ZCI6InYiLCJzdWIiOiJySWpsZ2RiaDAxZ2owMFFsTGI1WVNFWVJIanJsYms2U3JNQjJTZHpEa2N4WW8ifQ.QBUPkiaZSrRFnLOXvm0eOGAP-GhmW_MCwPCfJhaPpCYf5-diwgi6cSQROsC7LIrnvPhb0vdDBPmx8qslr-bcNMyjJ808_GainSww0nUH1yy409QrRV-Wvdrmw6O0lhGIhNTQ8M9SCBA9Cb9ENvjWwqQQW0RMM0YW4w9-_oAGyBwWhLS6j_WOXrOVPP0NdYiPq_3gk3Qwl8ov8pgVe17HdZYBTPNfY33p9EjhBSu6iueK_GIi7qC3qJ6b_N7uJ0HX5Pk0fj85fpRpEY3SVoB62EKC4vc37KB0xfHtY6ybkJEGqzexZRQ8xfIsLyUQUIQpkO2PjZnJ4bDysVehfgtFJA%22%2C%22storyboard%22%3A%22eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImNCMjRZNUdiQm1DeXZTS2hEOW5FZXhaZDMzZ0lTZk80VVUzV2Q5cXM0aXcifQ.eyJleHAiOjE2ODM3NTUwMDEsImF1ZCI6InMiLCJzdWIiOiJySWpsZ2RiaDAxZ2owMFFsTGI1WVNFWVJIanJsYms2U3JNQjJTZHpEa2N4WW8ifQ.tQcqGWBp5ItcZ8JtwrbQvKSNBdzh0rYGrqmo8-PzpyqPVqtzKX-Y-8KVLAHU0Jv_7aZHu0mO6FBSXnBLBXPWrpirER11Gu9RXR_bO15SGGlezsvLVnneRQbiTzNt6_xHr6lSuBbJaoHRnn4HlLkKRP6vZiyzaws7vDkZ_xDDOW2GP1s-FXnXuC-sYLzHjpN0NgX961vuV1yJF8ZWuIawyaat_oTqkmSoAzSvib03DluaXWYbofqy_rYXJAkh4uxF_eoVtQNiZY1tBu_KQdbjnMHVRLMBChCKiw2HrMvs5AK0kW_ZoLZIwual-DgIvjh1frUacgNgfm966MaBuAdNBA%22%2C%22thumbnail%22%3A%22eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImNCMjRZNUdiQm1DeXZTS2hEOW5FZXhaZDMzZ0lTZk80VVUzV2Q5cXM0aXcifQ.eyJleHAiOjE2ODM3NTUwMDEsImF1ZCI6InQiLCJzdWIiOiJySWpsZ2RiaDAxZ2owMFFsTGI1WVNFWVJIanJsYms2U3JNQjJTZHpEa2N4WW8ifQ.HVTJXkgdL8zmavm6gwTpzvDD-Rp8V0muZ7uuEMEpvM8SOEp3FE8MTWey51A-uZwR2CXD22y_6A6jTUhrabgXigYoyc3NciCiLzi3JpPFWnACfN12YJYnyst_O6ptk-YysyGyaQS4hZrNOc-G-HwRMe0zCwkNaVRyCsXFbTiXH-y9mJ6LDDlPS1Dq-PNs8y7f-NMBmBgjrYCim0ji5HkMpzOw9FbOlc_kgEukgrPxXIJFIFL8ngsrYcWpjtrMt-LG73jh7q5SJsWBZwdATIgjjF7xyuk_JKPiZSaIaM2W70ZIy1ONGkDpF6GjodzsFnTYcnryDpJqBpOI9ifb_OHHZA%22%7D&playbackId=%22rIjlgdbh01gj00QlLb5YSEYRHjrlbk6SrMB2SdzDkcxYo%22).

**Comment by gkatsev** [comment]
We'll continue investigating, but we don't have a specific fix right now.

**Comment by sharpsteelsoftware1** [comment]
Thank you, that enabled to continue my development process. Great support over there at Mux.

**Comment by KrustyC** [comment]
I a getting the exact same error, both when using mux-player-react and mux-video-react. This indeed work just fine in local, but once published to production the video does not work; it only loads the first second and then it stops, but I can see a growing list of Network requests similar to the one reported above. 

Unfortunately I can not "wait until the user plays the video" as I am trying to use the component as a background video which auto plays in loop. 

Do you guys know of any workaround for such scenario?

**Comment by dylanjha** [comment]
@KrustyC do you have an example we can look at?

I'm looking back at both of these examples and I'm no longer able to repro

**Comment by cornwe19** [comment]
Having the same issue as well. My only options so far have been to resort to MP4 playback (which fortunately we can get away with as our videos are small/short) or hacking in the `mux-player` HTML element directly by referencing the package from a CDN at page load time. Neither is ideal.

**Comment by cornwe19** [comment]
@dylanjha I can try and find some time to create a simple sample project in a bit, but this seems to happen in any react project packaged for production based on my experience so far.

Repro steps should be as straightforward as create a new NextJS project, install either `@mux/mux-player-react` or `@mux/mux-video-react`, add a MuxPlayer or MuxVideo component with the playbackId of any video and yarn build / start (not yarn dev).

This appears to even be happening to us in Sanity Studio using the Mux plugin.

For some additional context, it seems the first M3U8 chunk actually can't be loaded at all when built for production. When my videos do start playing putting an end to the infinite loop trying to load chunk 0, the first ~5 seconds of the video are skipped and not playable.

**Comment by dylanjha** [comment]
Thanks for bearing with me @cornwe19 & all.

I have a brand new NextJS app here: 

[- repo](https://github.com/dylanjha/fresh-next-app)

You can see there is only [1 commit](https://github.com/dylanjha/fresh-next-app/commit/fef40ff39960eac72172abc50bc1bdfe157d62a6) which adds Mux Player

I'm not seeing the issue on [the deployed site](https://fresh-next-app-alpha.vercel.app/). Anyone know what is different in my React app vs. the ones you have?

I _can_ experience this issue on a customer's site, so I know my browser / device is capable of reproducing it. I just don't know why the issue is not happening in my brand new NextJS app linked above.

**Comment by cornwe19** [comment]
@dylanjha thanks for taking a look back into this. I just messed around with the repo you posted and it looks like I can reproduce the issue in your minimal example with a different video. Try this playback ID `54C9Sybec1IHL6QuzQnjkHkWxi5ffjgx2TfxLMrw00zA`

**Comment by dylanjha** [comment]
@cornwe19 indeed, it is an issue specific to only certain assets.

cc:/ @KrustyC 

Happy to report we are seeing that this is fixed by https://github.com/muxinc/elements/pull/718

TLDR here for folks following along:

- A while back we had to switch to minified version of Hls.js: https://github.com/muxinc/elements/pull/550, this was to resolve a [Sveltkit Build issue](https://github.com/muxinc/elements/issues/541) -- or more specifically a Vite issue

This bug you reported happens under specific conditions:

- Minified Hls.js
- Next.js build step (build step is needed, because it doesn't happen in dev mode)
- Specific video asset where the presentation time and the media time of the first segment have a discrepancy

Normally, Hls.js is able to handle time mis-matches but for some reason with the combination of Hls.js & a Next.js production build this bug services when this first ts segment is transmuxed.

If you could help confirm the behavior in the demo apps, with your problematic playback ID:

Current Mux Player release:
- https://elements-demo-nextjs-3296d13ma-mux.vercel.app/MuxPlayer?playbackId=%2254C9Sybec1IHL6QuzQnjkHkWxi5ffjgx2TfxLMrw00zA%22

Patch from PR #718 
- https://elements-demo-nextjs-git-fork-cjpillsbury-fix-use-st-824afb-mux.vercel.app/MuxPlayer?playbackId=%2254C9Sybec1IHL6QuzQnjkHkWxi5ffjgx2TfxLMrw00zA%22

Anyone else can use those above links as well to confirm with their problematic playback IDs.

**Comment by cornwe19** [comment]
@dylanjha Excellent news. The patch does appear to be working for me as well. Thanks!

**Comment by dylanjha** [comment]
okay thanks @cornwe19 -- we have a canary release now with this patch: [1.11.3-canary.0-c45d462](https://www.npmjs.com/package/@mux/mux-player-react/v/1.11.3-canary.0-c45d462)

Repro in my fresh-next-app with your playback ID:

- https://github.com/dylanjha/fresh-next-app/tree/a654d26c3b9166e8cba873df4c0122e229a84f59
- https://fresh-next-5vv8n3ld2-dylanjha.vercel.app/

Canary build of Mux Player: 

- https://github.com/dylanjha/fresh-next-app/tree/34ad6e0dbade2c3de7e9fa5fc3bd5c9935805135
- https://fresh-next-58pa7q2y6-dylanjha.vercel.app/

We'll work on getting an official patch release here shortly

**Comment by dylanjha** [comment]
Happy to report this is fixed in 1.11.4

- https://github.com/muxinc/elements/releases/tag/%40mux%2Fmux-player-react%401.11.4
- https://github.com/muxinc/elements/releases/tag/%40mux%2Fmux-player%401.11.4

Closing for now, please let us know if you have any other issues related to this.

**Comment by KrustyC** [comment]
@dylanjha sorry I was on holiday and did not check into this. Btw I can confirm that this solved the issue for me too (also when using react-video-player)
